### PR TITLE
sl-100: Stripe integration

### PR DESCRIPTION
**Pull request by dimapopovych**
image
image
Failed to implement this with stripe subscriptions, so decided to do this via charges

### CODE REVIEW

**Review by andanf-e — APPROVED** [review]

File: backend/src/db/migrations/20230405083951_add_subscription_table.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { type Knex } from 'knex';
+
+const TABLE_NAME = 'subscriptions';
+
+const ColumnName = {
+  ID: 'id',
+  SUBSCRIPTION_END: 'subscription_end',
```
Comment (what1s1ove) [inline backend/src/db/migrations/20230405083951_add_subscription_table.ts]:
```suggestion
  SUBSCRIPTION_END: 'end_date',
```

File: backend/src/packages/users/user.entity.ts [hunk 2]
```diff
@@ -19,6 +19,10 @@ class UserEntity implements IEntity {
 
   private 'avatarUrl': string | null;
 
+  private 'subscriptionId': number | null;
+
+  private 'subscriptionEnd': string | null;
```
Comment (what1s1ove) [inline backend/src/packages/users/user.entity.ts]:
```suggestion
  private 'subscriptionEndDate': string | null;
```

File: frontend/.env.example [hunk 3]
```diff
@@ -5,3 +5,6 @@ VITE_APP_API_ORIGIN_URL=/api/v1
 
 # DEV
 VITE_APP_PROXY_SERVER_URL=http://localhost:3001
+
+# STRIPE
+VITE_APP_STRIPE_PUBLIC_KEY=[stripe-public-key]
```
Comment (what1s1ove) [inline frontend/.env.example]:
```suggestion
VITE_APP_STRIPE_PUBLIC_KEY=stripe-public-key
```

File: frontend/src/libs/components/checkout/styles.scss [hunk 4]
```diff
@@ -0,0 +1,37 @@
+/* stylelint-disable selector-class-pattern */
```
Comment (what1s1ove) [inline frontend/src/libs/components/checkout/styles.scss]: add overrides in `stylelint` config for this

File: frontend/src/libs/components/checkout/checkout.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+const Checkout: React.FC<Properties> = ({
+  onCheckout,
+  price,
+  label,
+}: Properties) => {
+  return (
+    <StripeCheckout
+      stripeKey={config.ENV.STRIPE.STRIPE_PUBLIC_KEY}
+      token={onCheckout}
+      amount={price * 100}
```
Comment (what1s1ove) [inline frontend/src/libs/components/checkout/checkout.tsx]: add constant

**Review by what1s1ove — CHANGES_REQUESTED** [review]
do not forget to change schema in readme.md

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**